ci/run_test_memcheck.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# This is the script used for the run_test_memcheck.sh stage in CI
+set -ex
+
+# JENKINS-52781 tar function is breaking symlinks
+
+# shellcheck disable=SC1091
+source ./.build_vars.sh
+rm -f "${SL_BUILD_DIR}/src/control/src/github.com/daos-stack/daos/src/control"
+mkdir -p "${SL_BUILD_DIR}/src/control/src/github.com/daos-stack/daos/src/"
+ln -s ../../../../../../../../src/control \
+  "${SL_BUILD_DIR}/src/control/src/github.com/daos-stack/daos/src/control"
+DAOS_BASE=${SL_PREFIX%/install*}
+rm -rf valgrind_memcheck_results
+NODE=${NODELIST%%,*}
+mydir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+
+# shellcheck disable=SC2029
+ssh "$SSH_KEY_ARGS" jenkins@"$NODE" "DAOS_BASE=$DAOS_BASE      \
+                                     HOSTNAME=$HOSTNAME        \
+                                     HOSTPWD=$PWD              \
+                                     SL_PREFIX=$SL_PREFIX      \
+                                     $(cat "$mydir/run_test_memcheck_node.sh")"
```

Comment:
do we really need entirely new scripts for these?   The only difference is in what we remove and what we ultimately end up running, no?   It seems like most of it is boiler plate stuff we are copying to multiple places.   Could we not handle the differences with command line arguments instead?

Reply:
As @jolivier23 has pointed out, the existing scripts that your new `ci/` scripts are duplicating should be refactored with parameters such that they can be used for both the existing and your new stage, given that they are almost complete duplicates.